app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,35 @@
+class PostsController < ApplicationController
+  def index
+    @user = User.find(params[:user_id])
+    @posts = @user.posts
+  end
+
+  def show
+    @user = User.find(params[:user_id])
+    @post = Post.find(params[:id])
+  end
+
+  def new
+    @user = User.find(params[:user_id])
+    @post = Post.new
+  end
+
+  def create
+    set_current_user
+    @post = @current_user.posts.build(post_params)
+    @post.comments_counter = 0
+    @post.likes_counter = 0
+
+    if @post.save
+      redirect_to user_post_path(@current_user.id, @post.id), notice: 'Post was successfully Created'
+    else
```

Comment:
- [ ] There is an error preventing the user from creating a POST and therefore I am unable to check the rest os the apps functionality. Kindly fix this/

ScreenCast 
[cinnamon-2024-01-27T003255+0200.webm](https://github.com/faisalakb/R_blog_app/assets/81932478/e8bd28dc-19fe-4dc6-9b34-0544ea129d08)